### PR TITLE
(GH-337) Update vscode-languageclient and vscode modules

### LINKED ISSUE #337: Update node dependencies
Need to update our deps

### PR DESCRIPTION
**Pull request by glennsarti**
This commit updates the vscode-languageclient node module to 5.1.0.  This is
needed for more recent LSP features.

This is in two commits so you can see the progression as the extension takes on major version bumps in the node module.

This also updates the vscode dev dependency to remove security vulnerabilities and to get any bug fixes.

Fixes #337

### CODE REVIEW

**Comment by glennsarti** [comment]
Testing:

- [ ] Lang Server starrts
- [ ] Autocomplete and Hover
- [ ] Node Graph
- [ ] Inline linting
- [ ] Format Document
- [ ] Debug
- [ ] PDK Commands

**Comment by jpogran** [comment]
This is technically #337 not a maint commit...

**Comment by glennsarti** [comment]
@jpogran Commit messages modified.

File: src/PuppetLanguageClient.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { ConnectionStatus } from "./interfaces";
 import { LanguageClient } from "vscode-languageclient/lib/main";
 import { PuppetStatusBar } from "./PuppetStatusBar";
 import { ILogger } from "./logging";
-import { ServerOptions, LanguageClientOptions } from "vscode-languageclient/lib/client";
+import { LanguageClientOptions } from "vscode-languageclient/lib/client";
```
Comment (jpogran) [inline src/PuppetLanguageClient.ts]: Nit: this can be one line if you remove lib/client. This can be done in a new PR

**Pull request merged**